lib/intern.js

```diff
@@ -0,0 +1,494 @@
+const Assert = require('assert')
+const Util = require('util')
+const Q = require('./qbuilder')
+const Uuid = require('uuid')
+
+
+const intern = {
+  asyncMethod(f) {
+    return function (msg, done) {
+      const seneca = this
+      const p = f.call(seneca, msg)
+
+      Assert('function' === typeof p.then &&
```

Comment:
this is checking for a Promise object "duck type", not an async function - is that intentional?

Reply:
Yes, it is intentional, so that custom Promise implementations may be used, if necessary.